### ISSUE TITLE
[Bug]: #9833 removed the ability for newgrfs to create unremovable houses

### ISSUE THREAD

**Issue by PikkaBird** [OpenTTD/OpenTTD#12968]
### Version of OpenTTD

20240924

### Expected result

Prior to #9833, it was possible for a newgrf to create a completely unremovable house, by setting prop 10 'LA rating decrease on removal' to a value higher than 1000. This was useful to protect, for example, historically important buildings placed in limited numbers - like churches or town halls.

### Actual result

#9833 removed this possibility, as LA rating is no longer taken into consideration when the LA attitude is set to "Permissive". 

I believe this is not in line with the intended function of the patch - to allow the player to always perform actions as if they had an Outstanding rating - as with LA control enabled, even that rating does not allow the player to remove these houses without using the magic bulldozer.

Houses with prop 10 set to > 1000 should not be ordinarily removable by the player, regardless of the LA attitude setting.

### Steps to reproduce

Test GRF is attached:
[unremovablehousedemo.zip](https://github.com/user-attachments/files/17176917/unremovablehousedemo.zip)

**Comment by 2TallTyler** [comment]
You are correct that this was not intended with #9833. I wonder if in fixing this, we should add a new error message to communicate "You cannot bulldoze this house. It's not the local authority's decision, do not report this as a bug." Wording suggestions welcome. 🙂

**Comment by PeterN** [comment]
Heh, just "Can't clear this area..." is the normal message.

**Comment by PikkaBird** [comment]
> Heh, just "Can't clear this area..." is the normal message.

That's what I was thinking. Same error you get if you try to clear an industry, lighthouse, etc.

**Comment by u7467549** [comment]
Hey, I would like to try to tackle this issue if possible.

**Comment by u7467549** [comment]
Hey, I want to tackle this issue if possible

**Comment by Ashhsqwoi** [comment]
Hey, I want to hackle this issue if possible

**Comment by michaei-ost** [comment]
Hey, can I tackle this issue?

**Comment by 2TallTyler** [comment]
Go for it!
